blocks/library/latest-posts/index.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * WordPress dependencies
+ */
+import Placeholder from 'components/placeholder';
+
+/**
+ * Internal dependencies
+ */
+import { registerBlockType } from '../../api';
+import { getLatestPosts } from './data.js';
+
+registerBlockType( 'core/latestposts', {
+	title: wp.i18n.__( 'Latest Posts' ),
+
+	icon: 'list-view',
+
+	category: 'rest-api',
+
+	defaultAttributes: {
+		poststoshow: 5,
+	},
+
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+
+			this.state = {
+				latestPosts: [],
+			};
+
+			const { poststoshow } = this.props.attributes;
+
+			getLatestPosts( poststoshow )
```

Comment:
Is there any reason to store the request in the state. I tend to use simple class properties when changes to this property don't have to trigger a rerender.

> However, I'm not sure how I can easily test/trigger componentWillUnmount. Do you have some idea? :)

yep, hard to test. I'd suggest throttling the network requests and removing the block before the response (and ensure no warnings are shown in the console)